api/types.go

```diff
@@ -159,6 +165,7 @@ func (m *Message) UnmarshalJSON(b []byte) error {
 
 type ToolCall struct {
 	Function ToolCallFunction `json:"function"`
+	ID       string           `json:"id"`
```

Comment:
This ID is a pass through from user to us - not really required yet but added for server-side management of tools